### PR TITLE
feat(thrift): unwrap struct for streaming type descriptor

### PR DESCRIPTION
**Pull request by Marina-Sakai**
#### What type of PR is this?

feat

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: thrift/idl.go [hunk 1]
```diff
@@ -371,104 +373,138 @@ func addFunction(ctx context.Context, fn *parser.Function, tree *parser.Thrift,
 
 	}
 
+	st, err := streaming.ParseStreaming(fn)
+	if err != nil {
+		return err
+	}
+	isStreaming := st.ClientStreaming || st.ServerStreaming
```
Comment (AsterDY) [inline thrift/idl.go:380]: what if a  method used both on `ClientStreaming` and `ServerUnary`? what's the behavior?

Comment (Marina-Sakai) [reply thrift/idl.go:380]: > method used both on ClientStreaming and ServerUnary

Do you mean what will happen when the streaming.mode is "client" in the thrift idl?
like `Response EchoClient (1: Request req) (streaming.mode="client")`

Comment (Marina-Sakai) [reply thrift/idl.go:380]: wait, I got what your question meant. Let me sort out

Comment (Marina-Sakai) [reply thrift/idl.go:380]: When the method is a streaming method (either of client streaming, server streaming, bidirectional streaming), both request and response descriptor should be unwrapped.
It doesn't matter with the mode of req and resp. It depends on if the method (function) is streaming or unary.

**Review by jayantxie — APPROVED** [review]

**Review by AsterDY — APPROVED** [review]

**Pull request merged**